analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/ArithmeticCalculator.cs

```diff
@@ -167,7 +167,7 @@ private static NumberConstraint CalculateRemainder(NumberConstraint left, Number
         // If the absolute value of the divisor is bigger than the absolute value of the dividend, the result is equal to the dividend =>
         // If the absolute value of every divisor is bigger than the absolute value of any dividend => resulting range == dividend range.
         // Otherwise, the result is between 0 and the absolute value of the divisor - 1 for positive dividends or in the same range multiplied by -1 for negative dividends.
-        if (left.CanBeNegative || left.Max < MinOfAbsoluteValues(right))
+        if (left.CanBeNegative || BigInteger.Abs(left.Max) < MinOfAbsoluteValues(right))
```

Comment:
it's more verbose but also more explicit to say that we are checking here for the absolut values.